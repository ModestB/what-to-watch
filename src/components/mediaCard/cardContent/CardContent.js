import React, { useState, useEffect } from "react";

import classes from "./CardContent.module.scss";

import CardTitle from "./cardTitle/CardTitle";
import CardText from "./cardText/CardText";

export const CardContent = ({ showTitle, showOverview, showDate }) => {
  const [date, setDate] = useState("");

  useEffect(() => {
    const showFullDate = new Date(showDate);
    setDate(showFullDate.getFullYear());
  }, [showDate]);

  return (
    <div className={`${classes.container}`}>
      <CardTitle showTitle={showTitle} />
      <CardText showOverview={showOverview} />
      <p className={`${classes.date}`}>{date}</p>
    </div>
  );
};

export default CardContent;
