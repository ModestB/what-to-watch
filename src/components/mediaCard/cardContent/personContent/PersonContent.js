import React from "react";

import { connect } from "react-redux";

import useRoute from "../../../../hooks/useRoute";

import KnowForList from "../../knowForList/KnowForList";
import classes from "./PersonContent.module.scss";

export const PersonContent = (props) => {
  const { isSinglePage } = useRoute();
  let content = null;
  let name = null;
  let bodyContent = null;

  name = (
    <div
      className={`${[
        classes.name,
        !isSinglePage ? classes.nameShort : null,
      ].join(" ")}`}
    >
      {props.personName}
    </div>
  );

  if (props.profileKnownFor) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <KnowForList elements={props.profileKnownFor} />
      </div>
    );
  }

  if (isSinglePage) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.known_for_department
            ? props.profileDetails.known_for_department
            : "-"}
        </p>
        <p className={`${classes.title}`}>Birthday </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.birthday ? props.profileDetails.birthday : "-"}
        </p>
        <p className={`${classes.title}`}>Place Of Birth</p>
        <p className={`${classes.text}`}>
          {props.profileDetails.place_of_birth
            ? props.profileDetails.place_of_birth
            : "-"}
        </p>
      </div>
    );
  }

  content = (
    <React.Fragment>
      {name}
      {bodyContent}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    profileDetails: state.profileDetails,
  };
};

export default connect(mapStateProps, null)(PersonContent);
