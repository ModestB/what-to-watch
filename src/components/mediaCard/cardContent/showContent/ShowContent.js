import React from "react";

import useRoute from "../../../../hooks/useRoute";

import classes from "./ShowContent.module.scss";

export const ShowContent = (props) => {
  const { isSinglePage } = useRoute();

  let content = null;
  let cardTitle = null;
  let cardText = null;
  let date = null;

  if (!isSinglePage) {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle.length > 40
          ? props.showTitle.substring(0, 40) + "..."
          : props.showTitle}
      </div>
    );
  } else {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle}
      </div>
    );
  }

  if (props.showOverview && !isSinglePage) {
    cardText = (
      <p className={`${classes.description}`}>
        {props.showOverview.substring(0, 100) + "..."}
      </p>
    );
  } else {
    cardText = (
      <p
        className={`${[classes.description, classes.descriptionOverview].join(
          " "
        )} customScroll`}
      >
        {props.showOverview}
      </p>
    );
  }

  if (props.mediaType === "movie") {
    date = <p className={`${classes.date}`}>Released Date: {props.showDate}</p>;
  } else {
    date = <p className={`${classes.date}`}>Air Date: {props.showDate}</p>;
  }

  content = (
    <React.Fragment>
      {cardTitle}
      {cardText}
      {date}
    </React.Fragment>
  );
  return content;
};

export default ShowContent;
