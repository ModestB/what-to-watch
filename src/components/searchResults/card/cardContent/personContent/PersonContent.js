import React from "react";

import { connect } from "react-redux";

import KnowForList from "../../knowForList/KnowForList";
import classes from "./PersonContent.module.scss";

export const PersonContent = (props) => {
  let content = null;
  let name = null;
  let bodyContent = null;

  name = (
    <div
      className={`${[
        classes.name,
        !props.displaySinglePage ? classes.nameShort : null,
      ].join(" ")}`}
    >
      {props.personName}
    </div>
  );

  if (props.profileKnownFor) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <KnowForList elements={props.profileKnownFor} />
      </div>
    );
  }

  if (props.displaySinglePage) {
    bodyContent = (
      <div>
        <p className={`${classes.title}`}>Known for </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.known_for_department
            ? props.profileDetails.known_for_department
            : "-"}
        </p>
        <p className={`${classes.title}`}>Birthday </p>
        <p className={`${classes.text}`}>
          {props.profileDetails.birthday ? props.profileDetails.birthday : "-"}
        </p>
        <p className={`${classes.title}`}>Place Of Birth</p>
        <p className={`${classes.text}`}>
          {props.profileDetails.place_of_birth
            ? props.profileDetails.place_of_birth
            : "-"}
        </p>
      </div>
    );
  }

  content = (
    <React.Fragment>
      {name}
      {bodyContent}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    displaySinglePage: state.displaySinglePage,
    profileDetails: state.profileDetails,
  };
};

export default connect(
  mapStateProps,
  null
)(PersonContent);
