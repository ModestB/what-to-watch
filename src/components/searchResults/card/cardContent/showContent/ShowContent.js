import React from "react";

import { connect } from "react-redux";

import classes from "./ShowContent.module.scss";

export const ShowContent = (props) => {
  let content = null;
  let cardTitle = null;
  let cardText = null;
  let date = null;

  if (!props.displaySinglePage) {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle.length > 40
          ? props.showTitle.substring(0, 40) + "..."
          : props.showTitle}
      </div>
    );
  } else {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle}
      </div>
    );
  }

  if (props.showOverview && !props.displaySinglePage) {
    cardText = (
      <p className={`${classes.description}`}>
        {props.showOverview.substring(0, 130) + "..."}
      </p>
    );
  } else {
    cardText = (
      <p
        className={`${[classes.description, classes.descriptionOverview].join(
          " "
        )} customScroll`}
      >
        {props.showOverview}
      </p>
    );
  }

  if (props.mediaType === "movie") {
    date = <p className={`${classes.date}`}>Released Date: {props.showDate}</p>;
  } else {
    date = <p className={`${classes.date}`}>Air Date: {props.showDate}</p>;
  }

  content = (
    <React.Fragment>
      {cardTitle}
      {cardText}
      {date}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    displaySinglePage: state.displaySinglePage,
  };
};

export default connect(
  mapStateProps,
  null
)(ShowContent);
