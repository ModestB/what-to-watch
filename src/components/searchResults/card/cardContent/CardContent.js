import React from "react";

import ShowContent from "./showContent/ShowContent";
import PersonContent from "./personContent/PersonContent";

import classes from "./CardContent.module.scss";

export const CardContent = (props) => {
  let content = (
    <div className={`${classes.container}`}>
      {props.cardType !== "person" ? (
        <ShowContent
          showTitle={props.showTitle}
          showOverview={props.showOverview}
          showMediaType={props.showMediaType}
          showDate={props.showDate}
        />
      ) : (
        <PersonContent
          personName={props.personName}
          profileKnownFor={props.profileKnownFor}
        />
      )}
    </div>
  );
  return content;
};

export default CardContent;
