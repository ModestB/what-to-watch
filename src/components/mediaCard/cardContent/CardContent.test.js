import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { CardContent } from "./CardContent";
import ShowContent from "./showContent/ShowContent";
import PersonContent from "./personContent/PersonContent";

configure({ adapter: new Adapter() });

describe("<CardContent />", () => {
  let wrapper;
  let props = {};

  beforeEach(() => {
    wrapper = shallow(<CardContent {...props} />);
  });

  it("should match snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should have <ShowContent />", () => {
    expect(wrapper.find(ShowContent)).toHaveLength(1);
  });

  it("should match snapshot than cardType = person", () => {
    wrapper.setProps({ cardType: "person" });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should have <PersonContent /> than cardType = person", () => {
    wrapper.setProps({ cardType: "person" });
    expect(wrapper.find(PersonContent)).toHaveLength(1);
  });
});
