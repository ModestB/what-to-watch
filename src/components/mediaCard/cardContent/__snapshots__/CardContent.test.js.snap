// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContent /> should match snapshot 1`] = `
"<div className=\\"container\\">
  <Connect(ShowContent) showTitle={[undefined]} showOverview={[undefined]} showMediaType={[undefined]} showDate={[undefined]} />
</div>"
`;

exports[`<CardContent /> should match snapshot than cardType = person 1`] = `
"<div className=\\"container\\">
  <Connect(PersonContent) personName={[undefined]} profileKnownFor={[undefined]} />
</div>"
`;
